dashboard/state_aggregator.py

```diff
@@ -190,6 +190,12 @@ def _filter(
                     ):
                         # Case insensitive match for string filter values.
                         match = datum[filter_column].lower() == filter_value.lower()
+                    elif isinstance(filter_value, str) and isinstance(
```

Comment:
I thought this should already be covered by the `_convert_filters_type` above? Or we need the manual matching here?

Reply:
from my testing, we do need it here.

Reply:
ah i think I know why, we have a bug in matching Optional column. Added an issue to track: https://github.com/ray-project/ray/issues/41132